src/NuGet.Protocol.Catalog/FileCursor.cs

```diff
@@ -42,13 +42,12 @@ public FileCursor(string path, ILogger<FileCursor> logger)
             }
         }
 
-        public Task SetAsync(DateTimeOffset value, CancellationToken token)
+        public async Task SetAsync(DateTimeOffset value, CancellationToken token)
         {
             var data = new Data { Value = value };
             var jsonString = JsonConvert.SerializeObject(data);
-            File.WriteAllText(_path, jsonString);
+            await File.WriteAllTextAsync(_path, jsonString, token);
```

Comment:
Also here: Maybe use the new `System.Text.Json`?

Reply:
Feel free to give it a shot :D

#neverupgrade

Reply:
:P